src/rust/engine/src/externs/mod.rs

```diff
@@ -37,14 +36,10 @@ pub fn equals(h1: &PyAny, h2: &PyAny) -> bool {
   // between non-equal types to avoid legacy behavior like `assert True == 1`, which is very
   // surprising in interning, and would likely be surprising anywhere else in the engine where we
   // compare things.
-  if h1.get_type() != h2.get_type() {
+  if h1.get_type().ne(h2.get_type()).unwrap() {
```

Comment:
Note that this is slightly different: != was comparing identity. The replacement is `is()`

Reply:
Ah, thank you!